constraint/pkg/client/drivers/dummy/dummy.go

```diff
@@ -0,0 +1,217 @@
+		if code.Engine != d.name {
+			continue
+		}
+		dummyCode = code
+		found = true
+		break
+	}
+	if !found {
+		return errors.New("SimplePolicy code not defined")
+	}
+
+	source, err := schema.GetSource(dummyCode)
+	if err != nil {
+		return err
+	}
+
+	d.mtx.Lock()
+	defer d.mtx.Unlock()
+	if d.errOnTemplateAdd {
+		return fmt.Errorf("%w: test error for add template", ErrTesting)
+	}
+	d.code[ct.GetName()] = source.RejectWith
+	return nil
+}
+
+// RemoveTemplate removes all Compilers and Constraints for templ.
+// Returns nil if templ does not exist.
+func (d *Driver) RemoveTemplate(ctx context.Context, ct *templates.ConstraintTemplate) error {
+	d.mtx.Lock()
+	defer d.mtx.Unlock()
+	if d.errOnTemplateRemove {
+		return fmt.Errorf("%w: test error for remove template", ErrTesting)
+	}
+	delete(d.code, ct.GetName())
+	delete(d.constraints, ct.GetName())
+	return nil
+}
+
+// AddConstraint adds Constraint to storage. Used to validate state in tests.
+func (d *Driver) AddConstraint(ctx context.Context, constraint *unstructured.Unstructured) error {
+	d.mtx.Lock()
+	defer d.mtx.Unlock()
+	if d.errOnConstraintAdd {
+		return fmt.Errorf("%w: test error for add constraint", ErrTesting)
+	}
+	kind := strings.ToLower(constraint.GroupVersionKind().Kind)
+	if _, ok := d.constraints[kind]; !ok {
+		d.constraints[kind] = make(map[string]*unstructured.Unstructured)
+	}
+
+	d.constraints[kind][constraint.GetName()] = constraint.DeepCopy()
+
+	return nil
+}
+
+// RemoveConstraint removes Constraint from Rego storage. Used to validate state in tests.
+func (d *Driver) RemoveConstraint(ctx context.Context, constraint *unstructured.Unstructured) error {
+	d.mtx.Lock()
+	defer d.mtx.Unlock()
+	if d.errOnConstraintRemove {
+		return fmt.Errorf("%w: test error for remove constraint", ErrTesting)
+	}
+	kind := strings.ToLower(constraint.GroupVersionKind().Kind)
+
+	// if missing, consider the constraint removed
+	if _, ok := d.constraints[kind]; !ok {
+		return nil
+	}
+
+	delete(d.constraints[kind], constraint.GetName())
+
+	return nil
+}
+
+// AddData should not be a thing that drivers handle.
+func (d *Driver) AddData(ctx context.Context, target string, path storage.Path, data interface{}) error {
+	return nil
+}
+
+// RemoveData should not be a thing that drivers handle.
+func (d *Driver) RemoveData(ctx context.Context, target string, path storage.Path) error {
+	return nil
+}
+
+func (d *Driver) Query(ctx context.Context, target string, constraints []*unstructured.Unstructured, review interface{}, opts ...drivers.QueryOpt) ([]*types.Result, *string, error) {
+	results := []*types.Result{}
+	for i := range constraints {
+		constraint := constraints[i]
+		result := &types.Result{
+			Msg:        fmt.Sprintf("rejected by driver %s: %s", d.name, d.code[strings.ToLower(constraint.GetObjectKind().GroupVersionKind().Kind)]),
+			Constraint: constraint,
+			// TODO: the engine should not determine the enforcement action -- that does not work with CEL KEP
```

Comment:
highlighting TODO

Reply:
LMK if you want to see this done in this PR, I figured it'd be better as a follow-on

Reply:
ping @sozercan

Reply:
i am okay to defer, just want to capture this and attach an issue so we don't forget

Reply:
Added #298